packages/yarnpkg-pnpify/sources/RawHoister.ts

```diff
@@ -0,0 +1,213 @@
+/**
+ * Package id - a number that uniquiely identifies both package and its dependencies.
+ * There must be package with id 0 - which contains all the other packages.
+ */
+export type PackageId = number;
```

Comment:
No, all packages should be numbered before passing package tree to hoister, due to runtime performance considerations. `number` as a key to Maps works faster then `string`

Reply:
If they are guaranteed to be adjacent number (because pre-processed before being fed to the raw hoister), then maybe using an array rather than a map would be even better?

Reply:
@arcanis Do you mean using object rather than a map? I have benchmarked object vs Map<number, any>, Map is about 2 times faster, probably because object properties are always strings....

Reply:
No I mean using a real array. If I understand correctly, you'll be doing a prepass to convert all the `LocatorHash` into numbers. If that's the case, then those numbers will all be adjacents (first encountered hash will be `0`, second encountered hash will be `1`, etc), right?

If that's the case then the one advantage of a map (sparse entries) vs an array is lost, since we already know there will be no sparse entries (maybe some because of the values you remove, but I don't think removing them will have a strong memory impact given how short-lived will be this processing pass anyway 🤔).

So basically:

``` ts
type PackageMap = PackageInfo[];

// Assuming
declare const packageMap: PackageMap;
declare const pkgId: PackageId;

// Then to obtain the info:
const pkgInfo = packageMap[pkgId];
```

Reply:
Ah, yes, sure, thanks!

Reply:
Hmm, actually package tree can be an Array as well, I guess?

Reply:
@arcanis Done. Both tree and package infos are Arrays now.